components/eam/src/physics/cam/physpkg.F90

```diff
@@ -275,17 +277,18 @@ subroutine phys_register
 
        ! NB: has to be after chem_register to use tracer names
        ! Fields for gas chemistry tracers
-       if (history_gaschmbudget .or. history_gaschmbudget_2D .or. history_gaschmbudget_2D_levels) then
+       if (history_gaschmbudget .or. history_gaschmbudget_2D .or. history_gaschmbudget_2D_levels &
+            .or. history_chemdyg_summary) then
          call chm_diags_inti_ac() ! to get aer_species
          do m = 1,gas_pcnst
             if (.not. any( aer_species == m )) then
               spc_name = trim(solsym(m))
-              if (history_gaschmbudget .or. history_gaschmbudget_2D_levels) then
+              if (history_gaschmbudget .or. history_gaschmbudget_2D_levels .or. history_chemdyg_summary) then
                 gas_ac_name(m) = 'ac_'//spc_name
                 call pbuf_add_field(gas_ac_name(m), 'global', dtype_r8, (/pcols,pver/), gas_ac_idx)
               end if
 
-              if (history_gaschmbudget_2D) then
+              if (history_gaschmbudget_2D .or. history_chemdyg_summary) then
                 gas_ac_name_2D(m) = 'ac_2D_'//spc_name
                 call pbuf_add_field(gas_ac_name_2D(m), 'global', dtype_r8, (/pcols/), gas_ac_idx)
               end if
```

Comment:
Can you clarify why `history_chemdyg_summary` is needed in both if-statements? This means when `history_chemdyg_summary` is true, both 3D and 2D diagnostic fields will be saved.